src/libs/Network/PersistedRequestsQueue.js

```diff
@@ -27,33 +27,11 @@ function process() {
     }
 
     const tasks = _.map(persistedRequests, request => processRequest(request)
-        .then((response) => {
-            if (response.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
-                NetworkEvents.getLogger().info('Persisted optimistic request needs authentication');
-            } else {
-                NetworkEvents.getLogger().info('Persisted optimistic request returned a valid jsonCode. Not retrying.');
-            }
-            NetworkEvents.triggerResponse(request, response);
-            PersistedRequests.remove(request);
```

Comment:
Sorry I missed this before – if a persisted optimistic request receives 407 because an auth token is expired, should we keep it in Onyx to be retried after re-authentication?

Reply:
It just works like any other request....

1. We try it once and it gets a 407 then we will try to reauthenticate
2. If you fail to reauthenticate then game over - you get logged out and Onyx storage is cleared.